src/dex/balancer-v2/balancer-v2.ts

```diff
@@ -436,12 +497,33 @@ export class BalancerV2
       const _from = wrapETH(from, this.network);
       const _to = wrapETH(to, this.network);
 
-      const allPools = this.getPools(_from, _to);
+      const poolsWithTokens = this.getPools(_from, _to);
+
+      // limit pools are IDS of pools we can use (preceded with BalancerV2_)
+      // poolsWithTokens contains pool data for pools with tokenIn/Out
+      // allowedPools contains pool data for pools with tokenIn/Out that are in limit list
       const allowedPools = limitPools
-        ? allPools.filter(({ address }) =>
-            limitPools.includes(`${this.dexKey}_${address.toLowerCase()}`),
-          )
-        : allPools;
+        ? poolsWithTokens.filter(pool => {
+            const id = pool.id.split(pool.poolType.toLowerCase())[0];
+            // VirtualPools must have all their internal pools in limitPools
+            if (this.eventPools.virtualBoostedPools[id]) {
+              for (let t of this.eventPools.virtualBoostedPools[id]
+                .mainTokens) {
+                if (
+                  !limitPools.includes(
+                    `${this.dexKey}_${t.linearPoolAddr.toLowerCase()}`,
+                  )
+                )
+                  return false;
+              }
+              return true;
+            } else {
+              return limitPools.includes(
+                `${this.dexKey}_${pool.address.toLowerCase()}`,
+              );
+            }
+          })
+        : poolsWithTokens;
```

Comment:
This doesn't seem correct! You should check for all the identifiers that you returned in `getPoolIdentidier` for the boosted pool. You are missing check for the main pool id and phantom pool id. 
Also if you use the boosted pools you must make sure you don't use the underlying pools.